plugins/flytekit-envd/flytekitplugins/envd/image_builder.py

```diff
@@ -82,8 +82,13 @@ def build():
     install.python_packages(name=[{python_packages}])
     install.apt_packages(name=[{apt_packages}])
     runtime.environ(env={env}, extra_path=['/root'])
-    config.pip_index(url="{pip_index}")
 """
+    if image_spec.pip_extra_index_url is None:
+        envd_config += f'    config.pip_index(url="{pip_index}")\n'
+    else:
+        pip_extra_index_url = "\\n                ".join(image_spec.pip_extra_index_url)
```

Comment:
nit 
```suggestion
        pip_extra_index_url = " ".join(image_spec.pip_extra_index_url)
```

Reply:
@pingsutw The  "\\n                " is necessary to make sure that the 2nd and all following extra-index-urls get properly indented. Otherwise pip does not read the pip.conf correctly